In [1]:
import pandas as pd

titanic_df = pd.read_csv("titanic_data.csv")

## EDA

#### Basic Info

In [2]:
titanic_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


A pandas DataFrame with **891 passenger records** and **12 features**, combining numerical and categorical data. It includes survival status, class, demographics, family info, and ticket details. Some columns like **Age, Cabin, and Embarked** have missing values, making preprocessing necessary before analysis.


##### Specific Information

In [4]:
titanic_df[["Age", "Fare"]].describe()

,Age,Fare
count,714.000000,891.000000
mean,29.699118,32.204208
std,14.526497,49.693429
min,0.420000,0.000000
25%,20.125000,7.910400
50%,28.000000,14.454200
75%,38.000000,31.000000
max,80.000000,512.329200


**Description:**
    
* Average age of passengers is approximately 29.7 years, indicating that most passengers were relatively young.
* Age distribution is fairly symmetric, as the mean (29.7) and median (28.0) are close.
* The minimum age is 0.42 years (about 5 months), and the oldest passenger is 80 years old.
* There are missing values in the Age column (714 entries vs. 891 for Fare), suggesting some age data is not available.

* For Fare, the mean is 32.20 and the median is 14.45 — a large gap indicating a right-skewed (positively skewed) distribution.
* Fare values range from 0 (free tickets) to over 512, showing a wide spread and presence of outliers or high-paying passengers.
* The high standard deviation (49.69) also supports the presence of large variability in fare prices.


In [5]:
titanic_df[["Age", "Fare"]].mean()

Age     29.699118
Fare    32.204208
dtype: float64

In [8]:
titanic_df[["Age", "Fare"]].max()

Age      80.0000
Fare    512.3292
dtype: float64

In [14]:
titanic_df.value_counts("Survived", normalize = True)

Survived
0    0.616162
1    0.383838
Name: proportion, dtype: float64

In [16]:
titanic_df.Embarked.value_counts(dropna = False)

Embarked
S      644
C      168
Q       77
NaN      2
Name: count, dtype: int64

In [21]:
titanic_df[titanic_df.Sex == "female"].value_counts(
    "Survived"
)

Survived
1    233
0     81
Name: count, dtype: int64

In [25]:
titanic_df.value_counts(["Embarked","Survived"])

Embarked  Survived
S         0           427
          1           217
C         1            93
          0            75
Q         0            47
          1            30
Name: count, dtype: int64

In [28]:
# Embarked form S, gender Male , Survival rate ?

titanic_df[
    (titanic_df.Embarked == "S") & (titanic_df.Sex == "male")
].value_counts("Survived")


Survived
0    364
1     77
Name: count, dtype: int64

In [33]:
pd.crosstab(
    titanic_df.Sex,
    titanic_df.Survived,
    margins = True,
    margins_name = "All"
)

Survived,0,1,All
Sex,,,
female,81,233,314
male,468,109,577
All,549,342,891


In [36]:
pd.crosstab(
    index = [titanic_df.Embarked, titanic_df.Pclass],
    columns = [titanic_df.Sex, titanic_df.Survived],
    margins=True
)

Sex             female      male       All
Survived             0    1    0    1     
Embarked Pclass                           
C        1           1   42   25   17   85
         2           0    7    8    2   17
         3           8   15   33   10   66
Q        1           0    1    1    0    2
         2           0    2    1    0    3
         3           9   24   36    3   72
S        1           2   46   51   28  127
         2           6   61   82   15  164
         3          55   33  231   34  353
All                 81  231  468  109  889